databricks/koalas/indexes.py

```diff
@@ -399,6 +400,138 @@ def copy(self, name=None):
+
+        >>> s.index.value_counts().sort_index()
+        1.0    1
+        2.0    1
+        3.0    2
+        4.0    1
+        Name: count, dtype: int64
+
+        **sort**
+
+        With `sort` set to `False`, the result wouldn't be sorted by number of count.
+
+        >>> s.index.value_counts(sort=True).sort_index()
+        1.0    1
+        2.0    1
+        3.0    2
+        4.0    1
+        Name: count, dtype: int64
+
+        **normalize**
+
+        With `normalize` set to `True`, returns the relative frequency by
+        dividing all values by the sum of values.
+
+        >>> s.index.value_counts(normalize=True).sort_index()
+        1.0    0.2
+        2.0    0.2
+        3.0    0.4
+        4.0    0.2
+        Name: count, dtype: float64
+
+        **dropna**
+
+        With `dropna` set to `False` we can also see NaN index values.
+
+        >>> s.index.value_counts(dropna=False).sort_index()  # doctest: +SKIP
+        1.0    1
+        2.0    1
+        3.0    2
+        4.0    1
+        NaN    1
+        Name: count, dtype: int64
+
+        Also support for MultiIndex.
+
+        >>> midx = pd.MultiIndex([['lama', 'cow', 'falcon'],
+        ...                       ['speed', 'weight', 'length']],
+        ...                      [[0, 0, 0, 1, 1, 1, 2, 2, 2],
+        ...                       [1, 1, 1, 1, 1, 2, 1, 2, 2]])
+        >>> s = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3], index=midx)
+        >>> s.index  # doctest: +SKIP
+        MultiIndex([(  'lama', 'weight'),
+                    (  'lama', 'weight'),
+                    (  'lama', 'weight'),
+                    (   'cow', 'weight'),
+                    (   'cow', 'weight'),
+                    (   'cow', 'length'),
+                    ('falcon', 'weight'),
+                    ('falcon', 'length'),
+                    ('falcon', 'length')],
+                   )
+
+        >>> s.index.value_counts().sort_index()
+        (cow, length)       1
+        (cow, weight)       2
+        (falcon, length)    2
+        (falcon, weight)    1
+        (lama, weight)      3
+        Name: count, dtype: int64
+
+        >>> s.index.value_counts(normalize=True).sort_index()
+        (cow, length)       0.111111
+        (cow, weight)       0.222222
+        (falcon, length)    0.222222
+        (falcon, weight)    0.111111
+        (lama, weight)      0.333333
+        Name: count, dtype: float64
+        """
+        idx_name = SPARK_INDEX_NAME_FORMAT(0)
+        sdf = self._kdf._sdf
+        sdf_count = sdf.groupBy(self._scol.alias(idx_name)).count()
+
+        if dropna:
+            sdf_count = sdf_count.dropna()
+        if sort:
+            sdf_count = sdf_count.sort('count', ascending=ascending)
+        if normalize:
+            sdf_count = sdf_count.select(
+                idx_name,
+                (sdf_count['count'] / sdf_count.select(F.sum(sdf_count['count']))
+                                               .first()[0]).alias('count'))
```

Comment:
@itholic can't we do this in a single pass without execution job here? e.g. `sdf_count['count'] / F.sum(sdf_count['count'])`

Reply:
ping @itholic

Reply:
thanks for the comment! 

i tried, but they raises an exception like below:

```python
>>> sdf_count.select(sdf_count['count'] / F.sum(sdf_count['count']))
Traceback (most recent call last):
...
pyspark.sql.utils.AnalysisException: "grouping expressions sequence is empty, and '`count`' is not an aggregate function. Wrap '((CAST(`count` AS DOUBLE) / CAST(sum(`count`) AS DOUBLE)) AS `(count / sum(count))`)' in windowing function(s) or wrap '`count`' in first() (or first_value) if you don't care which value you get.;;\nAggregate [(cast(count#54L as double) / cast(sum(count#54L) as double)) AS (count / sum(count))#304]\n+- Aggregate [named_struct(__index_level_0__, __index_level_0__#0, __index_level_1__, __index_level_1__#1)], [named_struct(__index_level_0__, __index_level_0__#0, __index_level_1__, __index_level_1__#1) AS __index_level_0__#49, count(1) AS count#54L]\n   +- LogicalRDD [__index_level_0__#0, __index_level_1__#1, 0#2], false\n"
```

i think maybe if we want to code like that way, we should use crossJoin before like below:

```
sdf_count = sdf_count.crossJoin(
    sdf_count.select(
        F.sum(sdf_count['count']).alias('count_sum')))

sdf_count.select(F.col('count') / F.col('count_sum')).show()
+-------------------+
|(count / count_sum)|
+-------------------+
| 0.2222222222222222|
| 0.1111111111111111|
| 0.3333333333333333|
| 0.1111111111111111|
| 0.2222222222222222|
+-------------------+
```

how about this? i'm not pretty sure what solution is more efficient.

Reply:
Ah, actually there's `Series.value_counts` implementation. Can we move `Series.value_counts` into `IndexOpsMixin` and reuse?

Reply:
@HyukjinKwon oh i see. i fixed them to reuse Series one 👍